crypto/bio/b_sock.c

```diff
@@ -138,6 +161,15 @@ int BIO_get_host_ip(const char *str, unsigned char *ip)
     if (i > 0)
         return (1);
 
+    /* if gethostbyname_r is supported, use it. */
+# ifdef HAVE_GETHOSTBYNAME_R
+    he = &hostent;
+    err = BIO_gethostbyname_r(str, he, &result);
+    if ((err == 0) || (result == NULL)) {
```

Comment:
We don't normally have the inner parenthesis.  I.e. `if (err == 0 || result == NULL)`